graphix/transpiler.py

```diff
@@ -8,15 +8,43 @@
 
 import dataclasses
 from copy import deepcopy
-from typing import Optional, Sequence
-
+from typing import Optional, Sequence, List, Tuple
 import numpy as np
 
 import graphix.pauli
 import graphix.sim.base_backend
 from graphix.ops import Ops
 from graphix.pattern import Pattern
 from graphix.sim.statevec import Statevec
+from graphix import command
+from graphix.command import N, M, E, X, Z, CommandKind
+from graphix import instruction
+
+
+@dataclasses.dataclass
```

Comment:
Let's use either `pydantic.BaseModel` or `dataclass`, not both.

They have many in common.